### ISSUE TITLE
MD031 and MD076 conflict when code block is inside a list item

### ISSUE THREAD

**Issue by alies-dev** [rvben/rumdl#461]
## Description

MD031 (blank lines around fenced code blocks) and MD076 (blank line consistency between list items) conflict when a fenced code block appears inside an ordered list item.

## Steps to reproduce

Given this Markdown:

```markdown
1. First item
2. Second item with code block:

   ```json
   { "key": "value" }
   ```

3. Third item
4. Fourth item
```

- **MD076** reports: `Unexpected blank line between list items` (on the blank line between the closing ` ``` ` and item 3)
- Removing that blank line to fix MD076 then triggers **MD031**: `No blank line after fenced code block`
- MD031's auto-fix adds the blank line back, which would re-trigger MD076

## Expected behavior

The two rules should be aware of each other. When a fenced code block ends inside a list item, the required blank line after the code block (MD031) should not be flagged as an unexpected blank line between list items (MD076).

## Environment

- rumdl version: 0.1.26
- OS: macOS (arm64)
- Installed via: Homebrew

**Comment by rvben** [comment]
This was fixed in v0.1.27 (commit `25f8b8dc`). MD076 now recognizes blank lines required by MD031 (around fenced code blocks) and MD058 (around tables) as structural rather than inter-item separators.

Upgrading to v0.1.27 should resolve the conflict. Please reopen if you still see the issue after upgrading.

**Comment by stevehipwell** [comment]
@rvben I'm very impressed with the development speed, I came here to report this issue only to find it'd been resolved already.

**Comment by rvben** [comment]
Thanks @stevehipwell, I am trying my best to squash the bugs whenever they pop up. 😄

**Comment by sisp** [comment]
Thanks for creating this awesome tool, @rvben! ❤️

I just stumbled over this issue with a similar problem still present in v0.1.28.

Consider the following Markdown file:

````markdown
1. First item

1. Second item with code block:

    ```json
    {"key": "value"}
    ```

1. Third item
````

I'm getting the following linter error:

```console
$ rumdl --version
rumdl 0.1.28

$ rumdl check test.md
test.md:9:1: [MD076] Missing blank line between list items
```

**Comment by stevehipwell** [comment]
@sisp check out the indentation on your code block, I was seeing this behaviour as I had a code block indented to 4 spaces instead of 2.

**Comment by sisp** [comment]
@stevehipwell I neglected to mention that this also occurs when using the MkDocs flavor which renders the code block outside the list item when using only 2 spaces for indentation.
